client/src/modules/inventory/list/modals/log.modal.js

```diff
@@ -0,0 +1,85 @@
+angular.module('bhima.controllers')
+  .controller('InventoryLogModalController', InventoryLogModalController);
+
+InventoryLogModalController.$inject = [
+  'data', '$uibModalInstance', 'InventoryService',
+];
+
+function InventoryLogModalController(data, Instance, Inventory) {
+  const vm = this;
+  vm.close = Instance.close;
+
+  Inventory.read(data.uuid).then(inventory => {
+    vm.inventory = inventory;
+  });
+
+
+  Inventory.inventoryLog(data.uuid).then(logs => {
+    vm.logs = [];
+    logs.forEach(log => {
+      const { action, last, current } = JSON.parse(log.text);
+      formatKeys(last);
+      formatKeys(current);
+
+      if (action === 'CREATION') {
+        vm.logs.push({
+          value : 'FORM.INFO.CREATED',
+          date : log.log_timestamp,
+          userName : log.userName,
+        });
+      } else {
+        const updatedKeys = Object.keys(current);
+        updatedKeys.forEach(col => {
+          vm.logs.push({
+            col : Inventory.columnsMap(col),
+            value : getValue(last, current, col),
+            date : log.log_timestamp,
+            userName : log.userName,
+            update : true,
+          });
+        });
+      }
+    });
+
+  });
+
+
+  function formatKeys(record) {
+    const removables = ['group_uuid', 'type_id', 'unit_id'];
+    removables.forEach(r => {
+      delete record[r];
+    });
+    if (record.text) {
+      record.text = record.label;
+    }
+    return record;
+  }
+
+  function getValue(last, current, key) {
+    const result = {};
+
+    if (key === 'inventoryGroup') {
```

Comment:
This looks like it should be a `switch()` statement.  Can you make it into a switch statement with a `default` case?